go/ops.go

```diff
@@ -94,7 +94,13 @@ func (op *InnerOp) Apply(child []byte) ([]byte, error) {
 
 // CheckAgainstSpec will verify the LeafOp is in the format defined in spec
 func (op *LeafOp) CheckAgainstSpec(spec *ProofSpec) error {
+	if op == nil || spec == nil {
```

Comment:
Do we really need to check if the pointer receiver is `nil`? If it's `nil` wouldn't an attempt to call this function already cause a nil pointer dereference panic?

Reply:
At least in the fuzzer and error tests that I ran, that condition was failing.

Reply:
Done and reverted but I advise against it because there is a path in which op.CheckAgainstSpec is invoked directly without checking if the value was nil or not.